src/features/applications/update-applications.test.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  UpdateApplicationInfoAction,
+  updateApplicationInfo,
+} from "./update-applications";
+import { createMockClient } from "../../test/mocks/create-mock-client";
+import { asClient } from "../../test/utils/discord-conversions";
+import { volunteerApplicationsEmbed } from "./update-applications.fixture";
+import { Client, ButtonStyle } from "discord.js";
+
+jest.mock("../../config", () => ({
+  applicationsChannelId: "999888777",
+}));
+
+// Mock the base class methods
+const mockCreateOrUpdateInstructions = jest.fn();
+const mockGetChannel = jest.fn();
+
+jest.mock("../../shared/action/instructions-ready-action", () => ({
+  InstructionsReadyAction: class {
+    createOrUpdateInstructions = mockCreateOrUpdateInstructions;
+    getChannel = mockGetChannel;
+  },
+}));
+
+jest.mock("../../shared/action/ready-action", () => ({
+  readyActionExecutor: jest.fn((action, options) => action.execute()),
+}));
```

Comment:
Not crazy about this. Probably a cleaner way to do this. I expect we will need to do it a lot, also it looks like tight coupling. Maybe some IoC could fix this.